variables.tf

```
@@ -0,0 +1,20 @@
+variable "location" {
+    type = string
+    description = "Azure resource location"
+}
+
+variable "law_id" {
+    type = string
+    description = "Resource ID of the central log analytics workspace"
+}
```

As you need the location of the LAW (at least i guess so from the code) it is more reasonalbe to just use an object
```suggestion
variable "log_analytics_workspace" {
  type = object({
    id                  = string
    name                = string
    resource_group_name = string
    location            = string
  })
  description = "Workspace use for Monitoring. XY will be added to workspace."
}
```